### PR TITLE
Change file analzyer to be a composition with TextStructure.

### PR DESCRIPTION
**Pull request by hzeller**
... instead of inheritance.

The only downside is that this requires the slightly ugly `RelinquishTextStructure()`. Maybe you have an idea.

### CODE REVIEW

File: common/analysis/file_analyzer.h [hunk 1]
```diff
@@ -144,7 +145,22 @@ class FileAnalyzer : public TextStructure {
...
+  // Convenience methods to access text structure view.
+  const ConcreteSyntaxTree& SyntaxTree() const {
+    return ABSL_DIE_IF_NULL(text_structure_)->SyntaxTree();
+  }
+  const TextStructureView& Data() const { return text_structure_->Data(); }
+  TextStructureView& MutableData() { return text_structure_->MutableData(); }
+
+  // Return text structure used in this analysis. This object must be
+  // considered invalid afterwards.
+  std::unique_ptr<TextStructure> RelinquishTextStructure() {
```
Comment (fangism) [inline common/analysis/file_analyzer.h:157]: Possible alternative names: `TakeTextStructure`, `ReleaseTextStructure`.

**Review by fangism — APPROVED** [review]
Good first step to cleaning up.

File: common/analysis/file_analyzer.h (hunk 1 above)
Comment (hzeller) [reply common/analysis/file_analyzer.h]: I like `ReleaseTextStructure`. changed.

File: common/analysis/file_analyzer.h [hunk 2]
```diff
@@ -144,7 +146,27 @@ class FileAnalyzer : public TextStructure {
...
+  const TextStructureView& Data() const {
+    return ABSL_DIE_IF_NULL(text_structure_)->Data();
+  }
+  TextStructureView& MutableData() {
+    return ABSL_DIE_IF_NULL(text_structure_)->MutableData();
+  }
+
+  // Return text structure used in this analysis.
+  // This FileAnalyzer object must be considered invalid afterwards.
+  // TODO(hzeller): drive decomposition further so that we don't need this wart.
```
Comment (hzeller) [inline common/analysis/file_analyzer.h:162]: I've added this TODO to revisit later.

**Pull request merged**